First pull request

demar01 opened the pull request:
Hi Jesse

These are the changes I made on PeCorA. I changed the README.md quite a bit; we can decide later what we want to say/show there.

jessegmeyerlab reviewed (changes requested):
Readmes:
	

- 

- [ ] change section 2 header to "install":

- [ ] add windows install instructions 
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;within R, development version
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;install.packages("devtools")
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;devtools::install_github("DeveloperName/PackageName") 	

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;within R, CRAN?
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;install.packages("PeCorA")

- [ ] update install location to jessegmeyerlab repo

- [ ] maybe under header 3, add subsection on how to use their own data, eg what columns/information are required

- [ ] Functions needed?

- &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;convert maxquant peptides.txt to pecora format 
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;would need user to input peptides.txt and unique group name from the LFQ col names
- &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;wrapper function to run via command line


loading into R saw this, is this a problem?:
> library(PeCorA)
Registered S3 methods overwritten by 'lme4':
  method                          from
  cooks.distance.influence.merMod car 
  influence.merMod                car 
  dfbeta.influence.merMod         car 
  dfbetas.influence.merMod        car

Testing vingette got error:
> scaled_peptides<-PeCorA_preprocessing(t, 8, 100, "cntrl")
Error in scale_by(ms1log2 ~ Condition * BioReplicate, t) : 
  could not find function "scale_by"
In addition: Warning message:
In which(is.na(as.numeric(t[, area_column_name])) == TRUE) :
  NAs introduced by coercion

demar01 commented:
> Readmes:
> 
> * [ ]  change section 2 header to "install":
Changed 
> * [ ]  add windows install instructions
> * within R, development version
>         install.packages("devtools")
>         devtools::install_github("jessegmeyerlab/PeCorA")
Changed
> * within R, CRAN?
>         install.packages("PeCorA")
Added that but still not submited to CRAN
> * [ ]  update install location to jessegmeyerlab repo
Changed
> * [ ]  maybe under header 3, add subsection on how to use their own data, eg what columns/information are required
Changed
> * [ ]  Functions needed?
> * convert maxquant peptides.txt to pecora format
>               would need user to input peptides.txt and unique group name from the LFQ col names
> * wrapper function to run via command line
Need to add those function to package
> loading into R saw this, is this a problem?:
> 
> > library(PeCorA)
> > Registered S3 methods overwritten by 'lme4':
> > method                          from
> > cooks.distance.influence.merMod car
> > influence.merMod                car
> > dfbeta.influence.merMod         car
> > dfbetas.influence.merMod        car

PeCorA package is loaded quietly in the Vignette now, so that message does not appear

> Testing vignette got error, does function need to be standardize::scale_by?:
> 
> > scaled_peptides<-PeCorA_preprocessing(t, 8, 100, "cntrl")
> > Error in scale_by(ms1log2 ~ Condition * BioReplicate, t) :
> > could not find function "scale_by"

library(standardize) is also added in the vignette now, which removes the error. standardize::scale_by is imported in PeCoRa package, but it doesn't seem to be imported correctly for now. 

> > In addition: Warning message:
> > In which(is.na(as.numeric(t[, area_column_name])) == TRUE) :
> > NAs introduced by coercion

I need to suppress that warning in PeCorA_preprocessing function.

jessegmeyerlab commented on R/PeCorA_preprocessing.R:
```diff
@@ -0,0 +1,65 @@
+
+  ## scale the data and then plot
+  t$ms1log2<-log2(as.numeric(t[,area_column_name]))
+  t$ms1scaled <- scale_by(ms1log2 ~ Condition*BioReplicate, t)
```
does this need to be changed to?
t$ms1scaled <- standardize::scale_by(ms1log2 ~ Condition*BioReplicate, t)

demar01 replied:
Changed

jessegmeyerlab commented on DESCRIPTION:
```diff
@@ -0,0 +1,35 @@
+    dplyr,
+    MSstats,
+    rlang,
+    standardize,
```
might need to move this to under "Depends"?

demar01 replied:
Moved to Depends

jessegmeyerlab reviewed (changes requested):
Great, thanks for doing this.  

Can you please update the install instructions in the vignette as well? Sorry that is what I meant when I suggested changing the second section to "install", I like "installation". I can change the main readme myself but less sure how to change the vingette afterwards. 

Can we please fix the dependency as well? I looked and maybe it needs to be under "depends" instead of "imports", or it might need to be called in the function as standardize::scale_by

see line comments related to above

Note it also does not correctly import ggplot

For the command line wrapper, please make it dump all the plots of significantly different peptides as before.

jessegmeyerlab commented on R/import_processed_data_for_PeCorA.R:
```diff
@@ -0,0 +1,63 @@
+  levels(as.factor(pmelt$variable))
+
+  pmelt$Condition = rep(0, nrow(pmelt))
+  pmelt$Condition[grep(pmelt$variable, pattern="_COVID19")]<-"COVID"
+  head(pmelt)
+  pmelt$Condition[grep(pmelt$variable, pattern="NON.COVID19")]<-"NONCO"
+  pmelt$Condition[grep(pmelt$variable, pattern="control")]<-"CNTRL"
```
Need to update this so the function can account for user-specific groups in their maxquant data.

demar01 replied:
Changed in function `import_processed_data_for_PeCorA_LFQ`

i.e.
t2 <- import_processed_data_for_PeCorA_LFQ( LFQfile = "peptides.txt",
condition1="CONTROL",
condition2="_COVID",
condition3="NON.COVID")

jessegmeyerlab reviewed (changes requested):
in addition to the other issues we discussed note the maxquant re-formatting function should be able to look for unique string in the LFQ columns specified by the user to define the quantification groups

demar01 commented:
> Great, thanks for doing this.
> 
> Can you please update the install instructions in the vignette as well? Sorry that is what I meant when I suggested changing the second section to "install", I like "installation". I can change the main readme myself but less sure how to change the vingette afterwards.

Vignette updated. See PeCorA_vignette.pdf

> Can we please fix the dependency as well? I looked and maybe it needs to be under "depends" instead of "imports", or it might need to be called in the function as standardize::scale_by

Changed now

> see line comments related to above
> 
> Note it also does not correctly import ggplot

It does import ggplot correctly now

> For the command line wrapper, please make it dump all the plots of significantly different peptides as before.

jessegmeyerlab commented:
Thanks for updating this. I just tried to test it and now it wont install at all. It might be some problem with the package dependencies "digest", "reshape", and/or "MSstats". 

`> devtools::install_github("demar01/pecora-1")
Downloading GitHub repo demar01/pecora-1@HEAD
These packages have more recent versions available.
It is recommended to update all of them.
Which would you like to update?

1: All                               
2: CRAN packages only                
3: None                              
4: digest   (0.6.25 -> 0.6.26) [CRAN]
5: quantreg (5.73   -> 5.74  ) [CRAN]

Enter one or more numbers, or an empty line to skip updates:3
√  checking for file 'C:\Users\jmeyer\AppData\Local\Temp\Rtmp0i44df\remotes6f5411803d1e\demar01-PeCorA-1-136091f/DESCRIPTION' ...
-  preparing 'PeCorA': (731ms)
√  checking DESCRIPTION meta-information ... 
-  checking for LF line-endings in source and make files and shell scripts
-  checking for empty or unneeded directories
     NB: this package now depends on R (>= 3.5.0)
     WARNING: Added dependency on R >= 3.5.0 because serialized objects in  serialize/load version 3 cannot be read in older versions of R.  File(s) containing such objects: 'PeCorA/data/Covid_peptides.rda'  'PeCorA/data/PeCorA_noZ.rda'  WARNING: Added dependency on R >= 3.5.0 because serialized objects in  serialize/load version 3 cannot be read in older versions of R.  File(s) containing such objects: 'PeCorA/data/input.dda.iprg.pg.rda'  WARNING: Added dependency on R >= 3.5.0 because serialized objects in  serialize/load version 3 cannot be read in older versions of R.  File(s) containing such objects: 'PeCorA/data/input.microglia.sk.rda'
-  building 'PeCorA_0.0.0.9000.tar.gz'
   
Installing package into ‘C:/Users/jmeyer/Documents/R/win-library/4.0’
(as ‘lib’ is unspecified)
* installing *source* package 'PeCorA' ...
** using staged installation
** R
** data
*** moving datasets to lazyload DB
** inst
** byte-compile and prepare package for lazy loading
Error: (converted from warning) package 'reshape' was built under R version 4.0.3
Execution halted
ERROR: lazy loading failed for package 'PeCorA'
* removing 'C:/Users/jmeyer/Documents/R/win-library/4.0/PeCorA'
* restoring previous 'C:/Users/jmeyer/Documents/R/win-library/4.0/PeCorA'
Error: Failed to install 'PeCorA' from GitHub:
  (converted from warning) installation of package ‘C:/Users/jmeyer/AppData/Local/Temp/Rtmp0i44df/file6f5473eb67f7/PeCorA_0.0.0.9000.tar.gz’ had non-zero exit status`

jessegmeyerlab commented:
Tried on my tablet to check if it was install specific and got a different error with MSstats?

`> devtools::install_github("demar01/pecora-1")
Downloading GitHub repo demar01/pecora-1@HEAD
These packages have more recent versions available.
It is recommended to update all of them.
Which would you like to update?

1: All                               
2: CRAN packages only                
3: None                              
4: digest   (0.6.25 -> 0.6.26) [CRAN]
5: rlang    (0.4.6  -> 0.4.8 ) [CRAN]
6: quantreg (5.73   -> 5.74  ) [CRAN]

Enter one or more numbers, or an empty line to skip updates:3
√  checking for file 'C:\Users\jesse\AppData\Local\Temp\Rtmpa80Mc0\remotese081b222ca\demar01-PeCorA-1-136091f/DESCRIPTION' (523ms)
-  preparing 'PeCorA': (2.6s)
√  checking DESCRIPTION meta-information ... 
-  checking for LF line-endings in source and make files and shell scripts
-  checking for empty or unneeded directories
     NB: this package now depends on R (>= 3.5.0)
     WARNING: Added dependency on R >= 3.5.0 because serialized objects in  serialize/load version 3 cannot be read in older versions of R.  File(s) containing such objects: 'PeCorA/data/Covid_peptides.rda'  'PeCorA/data/PeCorA_noZ.rda'  WARNING: Added dependency on R >= 3.5.0 because serialized objects in  serialize/load version 3 cannot be read in older versions of R.  File(s) containing such objects: 'PeCorA/data/input.dda.iprg.pg.rda'  WARNING: Added dependency on R >= 3.5.0 because serialized objects in  serialize/load version 3 cannot be read in older versions of R.  File(s) containing such objects: 'PeCorA/data/input.microglia.sk.rda'
-  building 'PeCorA_0.0.0.9000.tar.gz'
   
Installing package into ‘C:/Users/jesse/Documents/R/win-library/4.0’
(as ‘lib’ is unspecified)
ERROR: dependency 'MSstats' is not available for package 'PeCorA'
* removing 'C:/Users/jesse/Documents/R/win-library/4.0/PeCorA'
Error: Failed to install 'PeCorA' from GitHub:
  (converted from warning) installation of package ‘C:/Users/jesse/AppData/Local/Temp/Rtmpa80Mc0/filee087d506c6e/PeCorA_0.0.0.9000.tar.gz’ had non-zero exit status`

demar01 commented:
> Tried on my tablet to check if it was install specific and got a different error with MSstats?
> 
> `> devtools::install_github("demar01/pecora-1")
> Downloading GitHub repo demar01/pecora-1@HEAD
> These packages have more recent versions available.
> It is recommended to update all of them.
> Which would you like to update?
> 
> 1: All
> 2: CRAN packages only
> 3: None
> 4: digest (0.6.25 -> 0.6.26) [CRAN]
> 5: rlang (0.4.6 -> 0.4.8 ) [CRAN]
> 6: quantreg (5.73 -> 5.74 ) [CRAN]
> 
> Enter one or more numbers, or an empty line to skip updates:3
> √ checking for file 'C:\Users\jesse\AppData\Local\Temp\Rtmpa80Mc0\remotese081b222ca\demar01-PeCorA-1-136091f/DESCRIPTION' (523ms)
> 
> * preparing 'PeCorA': (2.6s)
>   √  checking DESCRIPTION meta-information ...
> * checking for LF line-endings in source and make files and shell scripts
> * checking for empty or unneeded directories
>   NB: this package now depends on R (>= 3.5.0)
>   WARNING: Added dependency on R >= 3.5.0 because serialized objects in  serialize/load version 3 cannot be read in older versions of R.  File(s) containing such objects: 'PeCorA/data/Covid_peptides.rda'  'PeCorA/data/PeCorA_noZ.rda'  WARNING: Added dependency on R >= 3.5.0 because serialized objects in  serialize/load version 3 cannot be read in older versions of R.  File(s) containing such objects: 'PeCorA/data/input.dda.iprg.pg.rda'  WARNING: Added dependency on R >= 3.5.0 because serialized objects in  serialize/load version 3 cannot be read in older versions of R.  File(s) containing such objects: 'PeCorA/data/input.microglia.sk.rda'
> * building 'PeCorA_0.0.0.9000.tar.gz'
> 
> Installing package into ‘C:/Users/jesse/Documents/R/win-library/4.0’
> (as ‘lib’ is unspecified)
> ERROR: dependency 'MSstats' is not available for package 'PeCorA'
>
> * removing 'C:/Users/jesse/Documents/R/win-library/4.0/PeCorA'
>   Error: Failed to install 'PeCorA' from GitHub:
>   (converted from warning) installation of package ‘C:/Users/jesse/AppData/Local/Temp/Rtmpa80Mc0/filee087d506c6e/PeCorA_0.0.0.9000.tar.gz’ had non-zero exit status`

I moved MSstats to dependency now. If I download it and run R CMD INSTALL PeCorA_0.0.0.9000.tar.gz it works for me. Does it for you? What if you update all the packages ( option 1)?

I made a wrapper to run PeCorA on the command line and updated the instructions in the vignette. It works on my computer, can you check on yours (windows?)

jessegmeyerlab commented:
Updating the packages didn't help, it seems related to the change to add those packages as "depends". I cloned your repo and will try to figure it out for windows and let you know. 

I moved those back out of depends to dissect the problem and now its complaining that its missing the function "import_processed_data_for_PeCorA_LFQ" 
from the R folder. 

Can you please check that you pushed that to the master? maybe the name is wrong

jessegmeyerlab commented:
Also not seeing the wrapper function - can you check that you pushed that to your fork?

demar01 reviewed (commented):
@jessegmeyerlab this is the wrapper function

demar01 commented:
> Updating the packages didn't help, it seems related to the change to add those packages as "depends". I cloned your repo and will try to figure it out for windows and let you know.
> 
> I moved those back out of depends to dissect the problem and now its complaining that its missing the function "import_processed_data_for_PeCorA_LFQ"
> from the R folder.
> 
> Can you please check that you pushed that to the master? maybe the name is wrong
I am  always posting to your master branch from mine: 
![image](https://user-images.githubusercontent.com/5008157/96454410-b6791800-1213-11eb-93c1-fce8fb9ff317.png)

> `** building package indices
> ** testing if installed package can be loaded from temporary location
> *** arch - i386
> Error: package or namespace load failed for 'PeCorA' in namespaceExport(ns, exports):
> undefined exports: import_processed_data_for_PeCorA_LFQ
> Error: loading failed
> Execution halted
> *** arch - x64
> Error: package or namespace load failed for 'PeCorA' in namespaceExport(ns, exports):
> undefined exports: import_processed_data_for_PeCorA_LFQ
> Error: loading failed
> Execution halted
> ERROR: loading failed for 'i386', 'x64'
> 
> * removing 'C:/Users/jmeyer/Documents/R/win-library/4.0/PeCorA'
>   Warning message:
>   In i.p(...) :
>   installation of package ‘C:/Users/jmeyer/AppData/Local/Temp/RtmpABcgkR/file4bfc1b3d72e5/PeCorA_0.0.0.9000.tar.gz’ had non-zero exit status`

jessegmeyerlab commented:
> @jessegmeyerlab this is the wrapper function

I see now thanks. I was confused because I didn't see it in the "R" folder. Is that OK to be in the directory?

jessegmeyerlab commented:
there is no "import_processed_data_for_PeCorA_LFQ" in the folder "R", but it is in the NAMESPACE. I think that is why it is complaining?

demar01 commented:
Lemme check

Maria Dermit
Postdoctoral Research Assistant | Centre for Molecular Oncology

Barts Cancer Institute - a Cancer Research UK Centre of Excellence
Queen Mary, University of London
John Vane Science Centre, Charterhouse Square, London EC1M 6BQ
Email: maria.dermit@qmul.ac.uk
________________________________
From: jessegmeyerlab <notifications@github.com>
Sent: Monday, October 19, 2020 2:13 PM
To: jessegmeyerlab/PeCorA <PeCorA@noreply.github.com>
Cc: Maria Dermit <maria.dermit@qmul.ac.uk>; Author <author@noreply.github.com>
Subject: Re: [jessegmeyerlab/PeCorA] First pull request (#2)


there is no "import_processed_data_for_PeCorA_LFQ" in the folder "R", but it is in the NAMESPACE. I think that is why it is complaining?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjessegmeyerlab%2FPeCorA%2Fpull%2F2%23issuecomment-712146946&data=04%7C01%7C%7C23a55dba72ed4ff7419408d87430bbaa%7C569df091b01340e386eebd9cb9e25814%7C0%7C0%7C637387099937202378%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=hS5WmWqzfBmszEmX2IdDfMc08pYv1BWmjOHfaKn7Oig%3D&reserved=0>, or unsubscribe<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FABGGWHK7LIJGYGH3PSU2X4TSLQ3OJANCNFSM4SQ7FJGQ&data=04%7C01%7C%7C23a55dba72ed4ff7419408d87430bbaa%7C569df091b01340e386eebd9cb9e25814%7C0%7C0%7C637387099937202378%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=AfRPvQilxqThaNHlhtjB1Th0eS73M10jT21EduQmELA%3D&reserved=0>.

demar01 commented:
> there is no "import_processed_data_for_PeCorA_LFQ" in the folder "R", but it is in the NAMESPACE. I think that is why it is complaining?

Correct, it should be there now. Is is working?

jessegmeyerlab commented:
Thanks! will check again now

jessegmeyerlab commented:
OK it worked on my computer that already had MSstats installed by devtools::install_github(), but failed on dependency MSstats not available when I didn't have it installed on my tablet. 

MSstats isn't on cran so does it make sense to have as a dependency? Which function is it required for?

jessegmeyerlab commented:
actually getting a lot of different errors now that MSstats error went away - there is some problem with package versions. I will see if I can figure it out this morning

demar01 commented:
You are right, we dont need MSstats. I remove it from the description. Lemme know if updating packages solve the problem.  Version  of packages uses is on the vignette - which I need to update to remove MSstats.

Maria Dermit
Postdoctoral Research Assistant | Centre for Molecular Oncology

Barts Cancer Institute - a Cancer Research UK Centre of Excellence
Queen Mary, University of London
John Vane Science Centre, Charterhouse Square, London EC1M 6BQ
Email: maria.dermit@qmul.ac.uk
________________________________
From: jessegmeyerlab <notifications@github.com>
Sent: Monday, October 19, 2020 2:38 PM
To: jessegmeyerlab/PeCorA <PeCorA@noreply.github.com>
Cc: Maria Dermit <maria.dermit@qmul.ac.uk>; Author <author@noreply.github.com>
Subject: Re: [jessegmeyerlab/PeCorA] First pull request (#2)


OK it worked on my computer that already had MSstats installed by devtools::install_github(), but failed on dependency MSstats not available when I didn't have it installed on my tablet.

MSstats isn't on cran so does it make sense to have as a dependency? Which function is it required for?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjessegmeyerlab%2FPeCorA%2Fpull%2F2%23issuecomment-712162744&data=04%7C01%7C%7Cef2800ab29a84e5eca6308d8743450d7%7C569df091b01340e386eebd9cb9e25814%7C0%7C0%7C637387115327394651%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=ye%2BHzHtlwcpNqPRQWA9%2FqFKcgz%2BTqVVrOdeolHxDdbA%3D&reserved=0>, or unsubscribe<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FABGGWHIP2E5DA6VP2WIRAWDSLQ6OTANCNFSM4SQ7FJGQ&data=04%7C01%7C%7Cef2800ab29a84e5eca6308d8743450d7%7C569df091b01340e386eebd9cb9e25814%7C0%7C0%7C637387115327404647%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=6l903q7BvLuG39zikOtHUvNhEFA5lgQO6rtscy2U1OA%3D&reserved=0>.

jessegmeyerlab commented:
OK got it to install now. Might have partially been issue with R version so I updated to 4.0.3 and ran it in the base R gui and it installed

going through the vingette and found this error in the control name
![image](https://user-images.githubusercontent.com/65188012/96464523-c75a6700-11dc-11eb-8068-3789af3f2281.png)

demar01 commented:
Great, we could write in the documentation:


PeCorA has been tested with R version 3.6.3. The following packages/version are required:

reshape_0.8.8 standardize_0.2.1  ggplot2_3.3.2

what do you mean error in the control name?



Maria Dermit
Postdoctoral Research Assistant | Centre for Molecular Oncology

Barts Cancer Institute - a Cancer Research UK Centre of Excellence
Queen Mary, University of London
John Vane Science Centre, Charterhouse Square, London EC1M 6BQ
Email: maria.dermit@qmul.ac.uk
________________________________
From: jessegmeyerlab <notifications@github.com>
Sent: Monday, October 19, 2020 3:29 PM
To: jessegmeyerlab/PeCorA <PeCorA@noreply.github.com>
Cc: Maria Dermit <maria.dermit@qmul.ac.uk>; Author <author@noreply.github.com>
Subject: Re: [jessegmeyerlab/PeCorA] First pull request (#2)


OK got it to install now. Might have partially been issue with R version so I updated to 4.0.3 and ran it in the base R gui and it installed

going through the vingette and found this error in the control name
[image]<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fuser-images.githubusercontent.com%2F65188012%2F96464523-c75a6700-11dc-11eb-8068-3789af3f2281.png&data=04%7C01%7C%7Cb630449010714d0998ae08d8743b6a92%7C569df091b01340e386eebd9cb9e25814%7C0%7C0%7C637387145834445860%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=yiZUqW1qHuXIA2SAfk37ysTi3WP6WiLyiA1qtJtVRwA%3D&reserved=0>

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjessegmeyerlab%2FPeCorA%2Fpull%2F2%23issuecomment-712200875&data=04%7C01%7C%7Cb630449010714d0998ae08d8743b6a92%7C569df091b01340e386eebd9cb9e25814%7C0%7C0%7C637387145834455862%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=R8lVmTEzy1o4UUlhF4Cuc0qV2e4yaDKJDF38sMWib0o%3D&reserved=0>, or unsubscribe<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FABGGWHPORNMDIAEZT346ABLSLRENHANCNFSM4SQ7FJGQ&data=04%7C01%7C%7Cb630449010714d0998ae08d8743b6a92%7C569df091b01340e386eebd9cb9e25814%7C0%7C0%7C637387145834465853%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=sTSWFPylr3n2PvjF3fufbsu0KWTny2gfUNCpFPsMNw8%3D&reserved=0>.

jessegmeyerlab commented:
Can you see in the image of the vingette? control_name="t<-..." when it should be control_name="cntrl"

> Great, we could write in the documentation: PeCorA has been tested with R version 3.6.3. The following packages/version are required: reshape_0.8.8 standardize_0.2.1 ggplot2_3.3.2 what do you mean error in the control name? Maria Dermit Postdoctoral Research Assistant | Centre for Molecular Oncology Barts Cancer Institute - a Cancer Research UK Centre of Excellence Queen Mary, University of London John Vane Science Centre, Charterhouse Square, London EC1M 6BQ Email: maria.dermit@qmul.ac.uk
> […](#)
> ________________________________ From: jessegmeyerlab <notifications@github.com> Sent: Monday, October 19, 2020 3:29 PM To: jessegmeyerlab/PeCorA <PeCorA@noreply.github.com> Cc: Maria Dermit <maria.dermit@qmul.ac.uk>; Author <author@noreply.github.com> Subject: Re: [jessegmeyerlab/PeCorA] First pull request (#2) OK got it to install now. Might have partially been issue with R version so I updated to 4.0.3 and ran it in the base R gui and it installed going through the vingette and found this error in the control name [image]<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fuser-images.githubusercontent.com%2F65188012%2F96464523-c75a6700-11dc-11eb-8068-3789af3f2281.png&data=04%7C01%7C%7Cb630449010714d0998ae08d8743b6a92%7C569df091b01340e386eebd9cb9e25814%7C0%7C0%7C637387145834445860%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=yiZUqW1qHuXIA2SAfk37ysTi3WP6WiLyiA1qtJtVRwA%3D&reserved=0> — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjessegmeyerlab%2FPeCorA%2Fpull%2F2%23issuecomment-712200875&data=04%7C01%7C%7Cb630449010714d0998ae08d8743b6a92%7C569df091b01340e386eebd9cb9e25814%7C0%7C0%7C637387145834455862%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=R8lVmTEzy1o4UUlhF4Cuc0qV2e4yaDKJDF38sMWib0o%3D&reserved=0>, or unsubscribe<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FABGGWHPORNMDIAEZT346ABLSLRENHANCNFSM4SQ7FJGQ&data=04%7C01%7C%7Cb630449010714d0998ae08d8743b6a92%7C569df091b01340e386eebd9cb9e25814%7C0%7C0%7C637387145834465853%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=sTSWFPylr3n2PvjF3fufbsu0KWTny2gfUNCpFPsMNw8%3D&reserved=0>.

c

jessegmeyerlab commented:
Also, I like that you added the math 
![image](https://user-images.githubusercontent.com/65188012/96464949-43ed4580-11dd-11eb-95ae-8cd84d5bfd22.png)

I'm not an expert in linear models so maybe you understand this better, but are the groups X1 and X2 the condition groups or the peptide groups? If the latter then please change "x1= peptide<sub>i</sub>" and  x2 = "all other peptides". 

If X1 and X2 are the groups then there would X3 when we have three groups?

jessegmeyerlab commented:
I tested the steps in the vignette and it all works! If you can fix those two minor vingette parts I will merge this!

demar01 commented:
> I tested the steps in the vignette and it all works! If you can fix those two minor vingette parts I will merge this!

Awesome! I just 

> Can you see in the image of the vingette? control_name="t<-..." when it should be control_name="cntrl"
> 
> > Great, we could write in the documentation: PeCorA has been tested with R version 3.6.3. The following packages/version are required: reshape_0.8.8 standardize_0.2.1 ggplot2_3.3.2 what do you mean error in the control name? Maria Dermit Postdoctoral Research Assistant | Centre for Molecular Oncology Barts Cancer Institute - a Cancer Research UK Centre of Excellence Queen Mary, University of London John Vane Science Centre, Charterhouse Square, London EC1M 6BQ Email: [maria.dermit@qmul.ac.uk](mailto:maria.dermit@qmul.ac.uk)
> > […](#)
> > ________________________________ From: jessegmeyerlab [notifications@github.com](mailto:notifications@github.com) Sent: Monday, October 19, 2020 3:29 PM To: jessegmeyerlab/PeCorA [PeCorA@noreply.github.com](mailto:PeCorA@noreply.github.com) Cc: Maria Dermit [maria.dermit@qmul.ac.uk](mailto:maria.dermit@qmul.ac.uk); Author [author@noreply.github.com](mailto:author@noreply.github.com) Subject: Re: [jessegmeyerlab/PeCorA] First pull request (#2) OK got it to install now. Might have partially been issue with R version so I updated to 4.0.3 and ran it in the base R gui and it installed going through the vingette and found this error in the control name [image]https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fuser-images.githubusercontent.com%2F65188012%2F96464523-c75a6700-11dc-11eb-8068-3789af3f2281.png&data=04%7C01%7C%7Cb630449010714d0998ae08d8743b6a92%7C569df091b01340e386eebd9cb9e25814%7C0%7C0%7C637387145834445860%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=yiZUqW1qHuXIA2SAfk37ysTi3WP6WiLyiA1qtJtVRwA%3D&reserved=0 — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHubhttps://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjessegmeyerlab%2FPeCorA%2Fpull%2F2%23issuecomment-712200875&data=04%7C01%7C%7Cb630449010714d0998ae08d8743b6a92%7C569df091b01340e386eebd9cb9e25814%7C0%7C0%7C637387145834455862%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=R8lVmTEzy1o4UUlhF4Cuc0qV2e4yaDKJDF38sMWib0o%3D&reserved=0, or unsubscribehttps://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FABGGWHPORNMDIAEZT346ABLSLRENHANCNFSM4SQ7FJGQ&data=04%7C01%7C%7Cb630449010714d0998ae08d8743b6a92%7C569df091b01340e386eebd9cb9e25814%7C0%7C0%7C637387145834465853%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=sTSWFPylr3n2PvjF3fufbsu0KWTny2gfUNCpFPsMNw8%3D&reserved=0.
> 
> c

Ups! I missed that. It is fixed on the vignette now.

demar01 commented:
> Also, I like that you added the math
> ![image](https://user-images.githubusercontent.com/65188012/96464949-43ed4580-11dd-11eb-95ae-8cd84d5bfd22.png)

> I'm not an expert in linear models so maybe you understand this better, but are the groups X1 and X2 the condition groups or the peptide groups? If the latter then please change "x1= peptidei" and x2 = "all other peptides".
> 
> If X1 and X2 are the groups then there would X3 when we have three groups?

I think it is correct how it is  like that. It is the simplify version how we wrote it on Fig2C (image below) 

![image](https://user-images.githubusercontent.com/5008157/96477280-37450d80-122e-11eb-8e4c-b2684f26eefd.png)

and the exact complete formula ( for an example peptide)  from
lm(subtmpdf$ms1adj ~ subtmpdf$Condition*subtmpdf$allothers) is:
α+ β1(subtmpdf$ConditionEtOH)+ β2(subtmpdf$ConditionLPS)+ β3(subtmpdf$allothersC[+57]QGTFSPEDNSIK_all)+ β4(subtmpdf$ConditionEtOH× subtmpdf$allothersC[+57]QGTFSPEDNSIK_all)+ β5(subtmpdf$ConditionLPS× subtmpdf$allothersC[+57]QGTFSPEDNSIK_all)+ϵ

So there will be as many B as combinations are

demar01 commented:
> Also, I like that you added the math
> ![image](https://user-images.githubusercontent.com/65188012/96464949-43ed4580-11dd-11eb-95ae-8cd84d5bfd22.png)
> 
> I'm not an expert in linear models so maybe you understand this better, but are the groups X1 and X2 the condition groups or the peptide groups? If the latter then please change "x1= peptidei" and x2 = "all other peptides".
> 
> If X1 and X2 are the groups then there would X3 when we have three groups?

> Also, I like that you added the math
> ![image](https://user-images.githubusercontent.com/65188012/96464949-43ed4580-11dd-11eb-95ae-8cd84d5bfd22.png)

> I'm not an expert in linear models so maybe you understand this better, but are the groups X1 and X2 the condition groups or the peptide groups? If the latter then please change "x1= peptidei" and x2 = "all other peptides".
> 
> If X1 and X2 are the groups then there would X3 when we have three groups?

I think it is correct how it is  like that. It is the simplify version how we wrote it on Fig2C (image below) 

![image](https://user-images.githubusercontent.com/5008157/96477280-37450d80-122e-11eb-8e4c-b2684f26eefd.png)

and the exact complete formula ( for an example peptide)  from
lm(subtmpdf$ms1adj ~ subtmpdf$Condition*subtmpdf$allothers) is:
α+ β1(subtmpdf$ConditionEtOH)+ β2(subtmpdf$ConditionLPS)+ β3(subtmpdf$allothersC[+57]QGTFSPEDNSIK_all)+ β4(subtmpdf$ConditionEtOH× subtmpdf$allothersC[+57]QGTFSPEDNSIK_all)+ β5(subtmpdf$ConditionLPS× subtmpdf$allothersC[+57]QGTFSPEDNSIK_all)+ϵ

So there will be as many B as combinations are

demar01 commented:
> I tested the steps in the vignette and it all works! If you can fix those two minor vingette parts I will merge this!

Awesome! Lemme know what you think about the formula, if you think we  need to change it in the vignette.

demar01 commented:
Awesome! Are we thinking of submitting the package to CRAN?

I just built a cute hexticker for the package 🙂

I will send later a new pull request to include it.



Maria Dermit
Postdoctoral Research Assistant | Centre for Molecular Oncology

Barts Cancer Institute - a Cancer Research UK Centre of Excellence
Queen Mary, University of London
John Vane Science Centre, Charterhouse Square, London EC1M 6BQ
Email: maria.dermit@qmul.ac.uk
________________________________
From: jessegmeyerlab <notifications@github.com>
Sent: Monday, October 19, 2020 7:09 PM
To: jessegmeyerlab/PeCorA <PeCorA@noreply.github.com>
Cc: Maria Dermit <maria.dermit@qmul.ac.uk>; Author <author@noreply.github.com>
Subject: Re: [jessegmeyerlab/PeCorA] First pull request (#2)


Merged #2<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjessegmeyerlab%2FPeCorA%2Fpull%2F2&data=04%7C01%7C%7C3f285348e4d848f550c008d8745a14be%7C569df091b01340e386eebd9cb9e25814%7C0%7C0%7C637387277545066907%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=0EoaExOplupcGmC4u07LOZm82A8Hn%2F88KPiDzJLlyWc%3D&reserved=0> into master.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjessegmeyerlab%2FPeCorA%2Fpull%2F2%23event-3894851045&data=04%7C01%7C%7C3f285348e4d848f550c008d8745a14be%7C569df091b01340e386eebd9cb9e25814%7C0%7C0%7C637387277545076902%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=o6usaiprwYuSrwU8kTEw%2BH5BEHW6LxkAcLw3Uj53JjM%3D&reserved=0>, or unsubscribe<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FABGGWHO3G7ZU2BUCUBXRKRLSLR6EJANCNFSM4SQ7FJGQ&data=04%7C01%7C%7C3f285348e4d848f550c008d8745a14be%7C569df091b01340e386eebd9cb9e25814%7C0%7C0%7C637387277545076902%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=tPlM%2FedaDligNYAasbQN8QlhJsWdN%2FMTEmiHCvNYrPo%3D&reserved=0>.

demar01 commented:
Hi Maria,

That is adorable I love it!

Yes we should submit to CRAN. Do you know how that works in case we need to update it in the future.

Best,
Jesse

From: Maria Dermit <maria.dermit@qmul.ac.uk>
Sent: Monday, October 19, 2020 1:47 PM
To: jessegmeyerlab/PeCorA <reply@reply.github.com>; Meyer, Jesse <jesmeyer@mcw.edu>
Subject: Re: [jessegmeyerlab/PeCorA] First pull request (#2)

ATTENTION: This email originated from a sender outside of MCW. Use caution when clicking on links or opening attachments.
________________________________
Awesome! Are we thinking of submitting the package to CRAN?

I just built a cute hexticker for the package 🙂

I will send later a new pull request to include it.



Maria Dermit
Postdoctoral Research Assistant | Centre for Molecular Oncology

Barts Cancer Institute - a Cancer Research UK Centre of Excellence
Queen Mary, University of London
John Vane Science Centre, Charterhouse Square, London EC1M 6BQ
Email: maria.dermit@qmul.ac.uk<mailto:maria.dermit@qmul.ac.uk>
________________________________
From: jessegmeyerlab <notifications@github.com<mailto:notifications@github.com>>
Sent: Monday, October 19, 2020 7:09 PM
To: jessegmeyerlab/PeCorA <PeCorA@noreply.github.com<mailto:PeCorA@noreply.github.com>>
Cc: Maria Dermit <maria.dermit@qmul.ac.uk<mailto:maria.dermit@qmul.ac.uk>>; Author <author@noreply.github.com<mailto:author@noreply.github.com>>
Subject: Re: [jessegmeyerlab/PeCorA] First pull request (#2)


Merged #2<https://urldefense.com/v3/__https:/eur01.safelinks.protection.outlook.com/?url=https*3A*2F*2Fgithub.com*2Fjessegmeyerlab*2FPeCorA*2Fpull*2F2&data=04*7C01*7C*7C3f285348e4d848f550c008d8745a14be*7C569df091b01340e386eebd9cb9e25814*7C0*7C0*7C637387277545066907*7CUnknown*7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0*3D*7C1000&sdata=0EoaExOplupcGmC4u07LOZm82A8Hn*2F88KPiDzJLlyWc*3D&reserved=0__;JSUlJSUlJSUlJSUlJSUlJSUlJSU!!H8mHWRdzp34!sejezBWOCcGBY5X_yyGKKF7mES-FxwQRkC3BdxJZMbKiv-CKH7INg5EPkJ6SNjpQ$> into master.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/eur01.safelinks.protection.outlook.com/?url=https*3A*2F*2Fgithub.com*2Fjessegmeyerlab*2FPeCorA*2Fpull*2F2*23event-3894851045&data=04*7C01*7C*7C3f285348e4d848f550c008d8745a14be*7C569df091b01340e386eebd9cb9e25814*7C0*7C0*7C637387277545076902*7CUnknown*7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0*3D*7C1000&sdata=o6usaiprwYuSrwU8kTEw*2BH5BEHW6LxkAcLw3Uj53JjM*3D&reserved=0__;JSUlJSUlJSUlJSUlJSUlJSUlJSUl!!H8mHWRdzp34!sejezBWOCcGBY5X_yyGKKF7mES-FxwQRkC3BdxJZMbKiv-CKH7INg5EPkJ4Dn9Wh$>, or unsubscribe<https://urldefense.com/v3/__https:/eur01.safelinks.protection.outlook.com/?url=https*3A*2F*2Fgithub.com*2Fnotifications*2Funsubscribe-auth*2FABGGWHO3G7ZU2BUCUBXRKRLSLR6EJANCNFSM4SQ7FJGQ&data=04*7C01*7C*7C3f285348e4d848f550c008d8745a14be*7C569df091b01340e386eebd9cb9e25814*7C0*7C0*7C637387277545076902*7CUnknown*7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0*3D*7C1000&sdata=tPlM*2FedaDligNYAasbQN8QlhJsWdN*2FMTEmiHCvNYrPo*3D&reserved=0__;JSUlJSUlJSUlJSUlJSUlJSUlJSU!!H8mHWRdzp34!sejezBWOCcGBY5X_yyGKKF7mES-FxwQRkC3BdxJZMbKiv-CKH7INg5EPkFO8wR7j$>.
